src/main/java/org/cryptacular/util/CertUtil.java

```diff
@@ -507,18 +507,19 @@ public static <T> T encodeCert(final X509Certificate certificate, final EncodeTy
    * such as the common name (CN), organizational unit (OU), organization (O), locality (L), state (ST),
    * country (C), and other attributes.
    *
-   * @param cert           The X.509 certificate from which to extract the subject DN.
-   * @param includeSpaces  A boolean flag indicating whether to include spaces in the subject DN string.
-   * @return The subject DN string of the X.509 certificate. If {@code includeSpaces} is true,
-   *         the DN string will be returned in its original format. If {@code includeSpaces} is false,
-   *         the DN string will be returned without spaces between attributes.
+   * @param cert   The X.509 certificate from which to extract the subject DN.
+   * @param format Controls whether the output contains spaces between attributes in the DN.
+   *               Use {@link X500PrincipalFormat#CANONICAL} to generate a DN with spaces after the commas separating
+   *               attribute-value pairs, {@link X500PrincipalFormat#RFC2253} for no spaces.
+   * @return The subject DN string of the X.509 certificate.
    *
    * @throws NullPointerException  If the provided certificate is null.
    */
-  public static String subjectDN(final X509Certificate cert, final boolean includeSpaces)
+  public static String subjectDN(final X509Certificate cert, final X500PrincipalFormat format)
   {
     final X500Principal subjectX500Principal = cert.getSubjectX500Principal();
-    return includeSpaces ? subjectX500Principal.toString() : subjectX500Principal.getName();
+    return X500PrincipalFormat.CANONICAL.equals(format) ?
+      subjectX500Principal.toString() : subjectX500Principal.getName();
```

Comment:
Is it better to use this from X509Principal.java
`    public String getName(String format) {
        if (format != null) {
            if (format.equalsIgnoreCase(RFC1779)) {
                return thisX500Name.getRFC1779Name();
            } else if (format.equalsIgnoreCase(RFC2253)) {
                return thisX500Name.getRFC2253Name();
            } else if (format.equalsIgnoreCase(CANONICAL)) {
                return thisX500Name.getRFC2253CanonicalName();
            }
        }
        throw new IllegalArgumentException("invalid format specified");
    }`

Reply:
None of those formats provide the "with spaces" form that you're after. I would choose a different name than `CANONICAL` because the output here differs severely from `X500Principal.CANONICAL`

Reply:
@serac any suggestion on the name? Will make changes and we can close this.

Reply:
I'm not sure what the concern is. While I don't love the name `CANONICAL`, it's clearly documented and provides a switch to produce a DN with spaces. That's really all I care about. I recommend we proceed with MR as-is.

Reply:
That sounds good, will push up the changes soon.